src/inputs/internal/ComboBoxBase.tsx

```diff
@@ -404,8 +404,8 @@ function getInputValue<O>(
   multiselect: boolean,
   nothingSelectedText: string,
 ) {
-  return selectedOptions.length === 1
-    ? getOptionLabel(selectedOptions[0])
+  return selectedOptions.length > 0
+    ? selectedOptions.map(getOptionLabel).join(", ")
```

Comment:
Question: I think this affects more than just the read-only field view? I know only showing the count is a pain point we have, but I'd be hesitant to push a change like this through since the read-only view is the specific problem we're having 🤔

Reply:
Yeah, this actually affects more than just the read-only mode, I'll change it so it only changes on read-only